src/test/java/fi/otavanopisto/kuntaapi/test/server/integration/ptv/ServiceLocationServiceChannelInTestsIT.java

```diff
@@ -0,0 +1,149 @@
+    
+    waitApiListCount("/organizations", 3);
+    waitApiListCount("/serviceLocationServiceChannels", TestPtvConsts.SERVICE_LOCATION_SERVICE_CHANNELS.length);
+    waitApiListCount("/services", TestPtvConsts.SERVICES.length);
+  }
+  
+  @Test
+  public void updateServiceLocationServiceChannelUnauthorized() throws JsonParseException, JsonMappingException, IOException, InterruptedException {
+    ServiceLocationServiceChannel kuntaApiResource = getServiceLocationServiceChannel(0, TestPtvConsts.SERVICE_LOCATION_SERVICE_CHANNELS.length);
+    
+    givenReadonly()
+      .body(kuntaApiResource)
+      .contentType(ContentType.JSON)
+      .put("/serviceLocationServiceChannels/{kuntaApiChannelId}", kuntaApiResource.getId())
+      .then()
+      .assertThat()
+      .statusCode(401);
+  }
+  
+  @Test
+  public void updateServiceLocationServiceChannelForbidden() throws JsonParseException, JsonMappingException, IOException, InterruptedException {
+    ServiceLocationServiceChannel kuntaApiResource = getServiceLocationServiceChannel(0, TestPtvConsts.SERVICE_LOCATION_SERVICE_CHANNELS.length);
+    
+    givenReadWrite()
+      .body(kuntaApiResource)
+      .contentType(ContentType.JSON)
+      .put("/serviceLocationServiceChannels/{kuntaApiChannelId}", kuntaApiResource.getId())
+      .then()
+      .assertThat()
+      .statusCode(403);
+  }
+  
+  @Test
+  public void updateServiceLocationServiceChannelUnchanged() throws JsonParseException, JsonMappingException, IOException, InterruptedException {
+    String ptvId = TestPtvConsts.SERVICE_LOCATION_SERVICE_CHANNELS[0];
+    String organizationId = getOrganizationId(0);
+    String kuntaApiChannelId = getServiceLocationChannelId(0, TestPtvConsts.SERVICE_LOCATION_SERVICE_CHANNELS.length);
+    
+    grantOrganizationPermission(AccessType.READ_WRITE, organizationId, ClientOrganizationPermission.UPDATE_SERVICE_CHANNELS);
+
+    ServiceLocationServiceChannel kuntaApiResource = getServiceLocationServiceChannel(0, TestPtvConsts.SERVICE_LOCATION_SERVICE_CHANNELS.length);
+    V7VmOpenApiServiceLocationChannelInBase ptvInResource = getPtvServiceChannelMocker().readEntity(AbstractPtvMocker.PTV_IN_API, ptvId, V7VmOpenApiServiceLocationChannelInBase.class);
+    V7VmOpenApiServiceLocationChannel ptvOutResource =  getPtvServiceChannelMocker().readEntity(AbstractPtvMocker.PTV_OUT_API, ptvId, V7VmOpenApiServiceLocationChannel.class);
+    
+    getPtvServiceChannelMocker().mockServiceLocationPut(ptvId, ptvOutResource);
+    
+    givenReadWrite()
+      .body(kuntaApiResource)
+      .contentType(ContentType.JSON)
+      .put("/serviceLocationServiceChannels/{kuntaApiChannelId}",kuntaApiChannelId)
+      .then()
+      .assertThat()
+      .statusCode(200);
+
+    getPtvServiceChannelMocker().verifyServiceLocation(ptvId, ptvInResource);
+  }
+  
+  @Test
+  public void updateServiceLocationServiceChannelChanges() throws JsonParseException, JsonMappingException, IOException, InterruptedException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)